### PR TITLE
Fix port script fails when DB has no backfilled events.

### LINKED ISSUE #8618: Port script fails when DB has no backfilled events.
```
Traceback (most recent call last):
  File "/home/twilight/temp/synapse/env/bin/synapse_port_db", line 632, in run
    await self._setup_events_stream_seqs()
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/synapse/storage/database.py", line 569, in runInteraction
    result = await self.runWithConnection(
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/synapse/storage/database.py", line 646, in runWithConnection
    return await make_deferred_yieldable(
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/twisted/enterprise/adbapi.py", line 306, in _runWithConnection
    compat.reraise(excValue, excTraceback)
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/twisted/python/compat.py", line 464, in reraise
    raise exception.with_traceback(traceback)
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/twisted/enterprise/adbapi.py", line 297, in _runWithConnection
    result = func(conn, *args, **kw)
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/synapse/storage/database.py", line 641, in inner_func
    return func(conn, *args, **kwargs)
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/synapse/storage/database.py", line 447, in new_transaction
    r = func(cursor, *args, **kwargs)
  File "/home/twilight/temp/synapse/env/bin/synapse_port_db", line 823, in r
    txn.execute(
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/synapse/storage/database.py", line 212, in execute
    self._do_execute(self.txn.execute, sql, *args)
  File "/home/twilight/temp/synapse/env/lib/python3.8/site-packages/synapse/storage/database.py", line 238, in _do_execute
    return func(sql, *args)
psycopg2.errors.InvalidParameterValue: RESTART value (-1) cannot be less than MINVALUE (1)

```

Fix is:

```diff
diff --git a/scripts/synapse_port_db b/scripts/synapse_port_db
index 6c7664ad4..9794d004f 100755
--- a/scripts/synapse_port_db
+++ b/scripts/synapse_port_db
@@ -818,7 +818,7 @@ class Porter(object):
                     "ALTER SEQUENCE events_stream_seq RESTART WITH %s", (next_id,)
                 )
 
-            txn.execute("SELECT -MIN(stream_ordering) FROM events")
+            txn.execute("SELECT GREATEST(-MIN(stream_ordering), 1) FROM events")
             curr_id = txn.fetchone()[0]
             if curr_id:
                 next_id = curr_id + 1

```

### PR DESCRIPTION
**Pull request by erikjohnston**
Fixes #8618

### CODE REVIEW

File: scripts/synapse_port_db [hunk 1]
```diff
@@ -820,7 +820,7 @@ class Porter(object):
                     "ALTER SEQUENCE events_stream_seq RESTART WITH %s", (next_id,)
                 )
 
-            txn.execute("SELECT -MIN(stream_ordering) FROM events")
+            txn.execute("SELECT GREATEST(-MIN(stream_ordering), 1) FROM events")
             curr_id = txn.fetchone()[0]
             if curr_id:
```
Comment (richvdh) [inline scripts/synapse_port_db:825]: when would this be falsey?

File: scripts/synapse_port_db [hunk 2]
```diff
@@ -820,7 +820,7 @@ class Porter(object):
                     "ALTER SEQUENCE events_stream_seq RESTART WITH %s", (next_id,)
                 )
 
-            txn.execute("SELECT -MIN(stream_ordering) FROM events")
+            txn.execute("SELECT GREATEST(-MIN(stream_ordering), 1) FROM events")
```
Comment (anoadragon453) [inline scripts/synapse_port_db:823]: I assume this does fix things, but what is the purpose of negating the minimum `stream_ordering` here? And wouldn't using `GREATEST` and `-` here mean that we almost always end up with 1 as it's greater?

Comment (erikjohnston) [reply scripts/synapse_port_db:823]: `stream_ordering` is negative for backfilled events, so if e.g. the min is -52, then `GREATEST(- -52 1)` is `52`, while if there are no backfilled events we may end up with a positive min so we get `GREATEST(-1, 1)` which is `1`.

File: scripts/synapse_port_db (hunk 1 above)
Comment (erikjohnston) [reply scripts/synapse_port_db:825]: Ah, yeah now we have a `GREATEST` it will never be null :+1:

File: scripts/synapse_port_db [hunk 3]
```diff
@@ -820,14 +820,12 @@ class Porter(object):
...
             curr_id = txn.fetchone()[0]
-            if curr_id:
-                next_id = curr_id + 1
-                txn.execute(
-                    "ALTER SEQUENCE events_backfill_stream_seq RESTART WITH %s",
-                    (next_id,),
-                )
+            next_id = curr_id + 1
```
Comment (richvdh) [inline scripts/synapse_port_db:825]: sooo.. if there are no backfilled events, the SELECT will return 1, so we'll set the RESTART value to 2. I mean, it's not the end of the world, but it doesn't sound right.

Comment (erikjohnston) [reply scripts/synapse_port_db:825]: I mean, it really doesn't matter? I can change it if you want, but we do the exact same thing from a fresh DB too

Comment (richvdh) [reply scripts/synapse_port_db:825]: we do? fair enough. Thought you might change the GREATEST to 0.  I don't feel strongly.

**Review by richvdh — APPROVED** [review]
lgtm

**Pull request merged**